lib/src/lib.rs

```diff
@@ -51,8 +51,27 @@ pub mod fmt_util;
 pub mod fsmonitor;
 #[cfg(feature = "git")]
 pub mod git;
+#[cfg(not(feature = "git"))]
+/// A dummy module that provides a no-op implementation of the functions in the
+/// `git` module.
+pub mod git {
+    /// Determine, by its name, if a remote refers to the special local-only
+    /// "git" remote that is used in the Git backend.
+    ///
+    /// This function always returns false if the "git" feature is not enabled.
+    pub fn is_special_git_remote(_remote: &str) -> bool {
+        false
+    }
+}
 #[cfg(feature = "git")]
 pub mod git_backend;
+#[cfg(not(feature = "git"))]
+/// A dummy module that provides a no-op implementation of the functions in the
+/// `git_backend` module.
+pub mod git_backend {
+    /// A dummy struct that represents the Git backend.
+    pub struct GitBackend;
+}
```

Comment:
Looks like this can be removed if we define the two `git_backend()` methods in `cli_util` only when `git` is active. We can actually inline `ReadonlyUserRepo::git_backend()` too